### PR TITLE
Refactor: Move Value Mappings to Grafana public instead of Grafana UI

### LINKED ISSUE #44236: Refactor value mapping to not be in grafana-ui package
Move value mapping editor to be in public directory instead of inside of grafana-ui package in order to allow us to run more experiments / tune up UX without. If it makes sense we can move this back into the grafana-ui package in the future, but as this is an internal editor that can apply to every panel it may make sense to be included in grafana core

### PR DESCRIPTION
**Pull request by nmarrs**
**What this PR does / why we need it**:
Move ValueMappingEditor from `grafana-ui` library to grafana core as module is intended to be internal.

Removed ValueMappingEditor code from `grafana-ui` as internally our runtime behavior should be the exact same. Will update documentation in our plugins migration guide notifying removal of module from the `grafana-ui` library after refactoring threshold editor as well in a separate PR.

Documentation changes #44341

**Which issue(s) this PR fixes**:

Fixes #44236

### CODE REVIEW

File: packages/grafana-ui/src/components/OptionsUI/mappings.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 
 import { FieldConfigEditorProps, ValueMapping, ValueMappingFieldConfigSettings } from '@grafana/data';
-import { ValueMappingsEditor } from '../ValueMappingsEditor/ValueMappingsEditor';
+import { ValueMappingsEditor } from '../../../../../public/app/features/dimensions/editors/ValueMappingsEditor/ValueMappingsEditor';
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/OptionsUI/mappings.tsx:4]: this will break webpack!

**Comment by ryantxu** [comment]
@mckn -- any thoughts on this?  We are looking to update ValueMappingEditor, and was surprised it s in `@grafana/ui` -- this is not really a component you could even add (in any useful way)

This PR maps our how we can move it out of `@grafana/ui`, but is idential.

@nmarrs added deprecation notices to a version that stays in the package -- but I don't know it that is helpful or just noise?

Threshold editor will soon have similar questions -- again, I don't know of any reasonable other than the default.  If people are referencing these from the registry they will still work (just loaded at runtime rather than in the static array)

File: packages/grafana-ui/src/components/OptionsUI/mappings.tsx [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
 
-import { FieldConfigEditorProps, ValueMapping, ValueMappingFieldConfigSettings } from '@grafana/data';
-import { ValueMappingsEditor } from '../ValueMappingsEditor/ValueMappingsEditor';
+import {
+  deprecationWarning,
+  FieldConfigEditorProps,
+  ValueMapping,
+  ValueMappingFieldConfigSettings,
+} from '@grafana/data';
+import { ValueMappingsEditor } from '../../../../../public/app/features/dimensions/editors/ValueMappingsEditor/ValueMappingsEditor';
```
Comment (torkelo) [inline packages/grafana-ui/src/components/OptionsUI/mappings.tsx:9]: an import from grafana ui to the app is not allowed this will likley cause issues when trying to build an publish this as standalone package

File: packages/grafana-ui/src/components/OptionsUI/mappings.tsx [hunk 3]
```diff
@@ -11,6 +16,12 @@ export class ValueMappingsValueEditor extends React.PureComponent<
   }
 
   render() {
+    deprecationWarning(
+      'mappings.tsx',
+      'grafana-ui/src/components/OptionsUI',
+      'public/app/features/dimensions/editors/ValueMappingsEditor'
+    );
```
Comment (torkelo) [inline packages/grafana-ui/src/components/OptionsUI/mappings.tsx:23]: this deprecation warning does not make much sense as plugins cannot use components in the app. I would just remove it

**Review by torkelo — COMMENTED** [review]
You can move the whole thing and leave nothing in grafana-ui, I doubt no plugins use it. But to be on the safe side we can add them to the runtime exported grafana-ui object here: https://github.com/grafana/grafana/blob/main/public/app/features/plugins/plugin_loader.ts#L47 that way no runtime issues will happen if there is a plugin that uses it.

File: packages/grafana-ui/src/components/OptionsUI/mappings.tsx (hunk 2 above)
Comment (nmarrs) [reply packages/grafana-ui/src/components/OptionsUI/mappings.tsx:9]: This is good to know, thanks!

**Comment by nmarrs** [comment]
> You can move the whole thing and leave nothing in grafana-ui, I doubt no plugins use it. But to be on the safe side we can add them to the runtime exported grafana-ui object here: https://github.com/grafana/grafana/blob/main/public/app/features/plugins/plugin_loader.ts#L47 that way no runtime issues will happen if there is a plugin that uses it.

Is there a way to determine if a plugin in the wild is using a given module? From a runtime perspective internally we should be fine (as we are including the same editor as a default panel editor inside `public/app/core/components/editors/registry.tsx`) We probably will need a make a note of potentially breaking changes for plugins with this refactor. In general though it would be nice if we had a resource / analytics that discerned a given module's usage

**Review by mckn — COMMENTED** [review]
Thanks for pinging about this 🙏🏻 

I just had a quick search in the meta data we have gathered about what dependencies are being used by plugins published to grafana-com and this isn't used by any of them. Which makes sense according to what both @torkelo and @ryantxu says.

To be on the safe side please add a row about this being removed in the plugins migration guide. You can do it within this PR. Just add a new section for the version this will be included in and state that this is being removed due to being an internal component.

https://github.com/grafana/grafana/blob/main/docs/sources/developers/plugins/migration-guide.md

**Comment by ryantxu** [comment]
We should likely also move the threshold editor out of grafana/ui with the same notice:
image

**Comment by nmarrs** [comment]
> We should likely also move the threshold editor out of grafana/ui with the same notice: image

I can work on this refactor next and include the change in the documentation

**Comment by nmarrs** [comment]
As part of these changes I'm assuming that ValueMappingEditor documentation found in `grafana-ui` docs [here](https://developers.grafana.com/ui/latest/index.html?path=/story/pickers-and-editors-valuemappingseditor--basic) will also be removed (given removal of .story file)

Please correct me if I wrong in this assumption and if something else needs to be updated in order to remove this reference from grafana-ui's documentation

File: docs/sources/developers/plugins/migration-guide.md [hunk 4]
```diff
@@ -33,6 +35,14 @@ This guide helps you identify the steps you need to take based on the Grafana ve
     - [Migrate to data frames](#migrate-to-data-frames)
   - [Troubleshoot plugin migration](#troubleshoot-plugin-migration)
 
+## From version 8.3.x to 8.4.x
+
+This section explains how to migrate Grafana v8.3.x plugins to the updated plugin system available in Grafana v8.4.x. Depending on your plugin, you need to perform one or more of the following steps.
+
+### Value Mapping Editor has been removed from @grafana-ui library
+
+Removed due to being an internal component.
```
Comment (ryantxu) [inline docs/sources/developers/plugins/migration-guide.md:44]: The text and messaging here needs some :heart: -- lets move any notes to its own PR.  This feels like something that should be as minor as possible since any usage would be pretty exceptional

Comment (ryantxu) [reply docs/sources/developers/plugins/migration-guide.md:44]: lets move the docs changes to its own PR and address it after doing the same with thersholds

**Review by ryantxu — APPROVED** [review]
Lets move the docs changes into its own PR -- but other wise this is good to go
